### ISSUE TITLE
x86: implement system calls

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#2027]
**_Reported by Andrew Boie:_**

Once we have defined the architecture-neutral aspects of how system calls will work in Zephyr, create the necessary implementation for x86. We will need to determine how to marshal arguments and return values, and perform the necessary privilege elevation (it will most certainly involve an 'int' instruction to a designated vector).

(Imported from Jira ZEP-2189)

**Comment by nashif** [comment]
**_by Andrew Boie:_**

I have this implemented:
https://github.com/andrewboie/zephyr/tree/userspace

Waiting on another PR to land before I make a PR for it.

**Comment by nashif** [comment]
**_by Andrew Boie:_**

https://github.com/zephyrproject-rtos/zephyr/pull/1409
